source/features/move-marketplace-link-to-profile-dropdown.tsx

```diff
@@ -1,25 +1,25 @@
 import React from 'dom-chef';
 import select from 'select-dom';
+import elementReady from 'element-ready';
+import domLoaded from 'dom-loaded';
 import features from '../libs/features';
 
-function init(): false | void {
-	const lastDivider = select.last('.user-nav .dropdown-divider');
-	if (!lastDivider) {
-		return false;
-	}
+async function init(): Promise<void> {
+	(await elementReady('.Header-link[href="/marketplace"]'))!.remove();
```

Comment:
Wouldn't it be better to do this *after* adding the replacement link in case it fails? That way, if Github were to change the class (again), the Marketplace link would at least remain accessible in its "normal" location.

Reply:
That's a good point, but we want to remove it before the first paint to avoid, the layout change on every page in the header.

Reply:
That's fair. Would restoring the constant with the `if / return` be a way to work around that? If `select.last(...)` is stored/checked first, theoretically the rest can remain the same with the main navbar link still being removed before `domLoaded`.

Reply:
There's a way to do what you're saying, but the way I set it up now will show an error in the console and it'll push people to fix the selector sooner. Before this PR, the error was swallowed and nobody saw it